src/main_calc.c

```diff
@@ -71,6 +76,20 @@ void main(void) {
     asm("ld iy, 0D00080h");
     asm("set 3, (iy+024h)");
 
+    // check if a program was used as input
+    ti_CloseAll();
+    ice.usingInputProgram = false;
+    inputProgram = os_RclAns(&inputProgramType);
+    if (inputProgram && inputProgramType == TI_STRING_TYPE && (inputProgramSize = *inputProgram) < 9) {
+        memset(var_name, 0, sizeof var_name);
+        memcpy(var_name, inputProgram + 2, inputProgramSize);
+        programSlot = ti_OpenVar(var_name, "r", TI_PRGM_TYPE);
+        ti_CloseAll();
```

Comment:
Looks like files get closed already 20 lines below, so this should be unnecessary.